### PR TITLE
scalability framework: add DDL operations

### PR DESCRIPTION
**Pull request by nrainer-materialize**
This closes https://github.com/MaterializeInc/materialize/issues/23799.

### Commits
95b64eab25 scalability framework: introduce SqlOperationWithInput
d99f419521 scalability framework: introduce SqlOperationWithSeed
bb315f7ef1 scalability framework: provide worker_id as input data
79cd90dba0 scalability framework: allow workloads to have init operations
a373f47cd5 scalability framework: add DDL operations
35c1442c6c scalability framework: add DDL workloads
1792891fdf scalability framework: remove alter-column operations and workloads
1e523d45ba scalability framework: separate build step for DDL workloads

### Nightlies
https://buildkite.com/materialize/nightlies/builds?branch=nrainer-materialize%3Ascalability%2Fddl

### CODE REVIEW

**Review by def- — APPROVED** [review]

**Comment by nrainer-materialize** [comment]
I will need to find an appropriate timeout. Though, I might give the DDL stuff a separate build.

**Comment by jkosh44** [comment]
Do we mock S3 by using `minio`? If so, one concern is that when we switch to the persist catalog, the number won't be that accurate since `minio` is much faster than S3.

**Comment by def-** [comment]
Yes, we do. We could simulate additional latency to minio. Would that help?

**Comment by jkosh44** [comment]
> Yes, we do. We could simulate additional latency to minio. Would that help?

Hm, I'm not sure, maybe? I don't think it has to block this PR, just something to think about.

**Review by jkosh44 — APPROVED** [review]
LGTM!

File: misc/python/materialize/scalability/operations.py [hunk 1]
```diff
@@ -9,51 +9,128 @@
...
 
+class CreateTableX(SqlOperationWithSeed):
+    def __init__(self) -> None:
+        super().__init__("table_seed")
+
+    def sql_statement(self, table_seed: str) -> str:
+        return f"CREATE TABLE x_{table_seed} (f1 INT, f2 INT, f3 INT, f4 INT, f5 INT);"
+
+
+class CreateIndexOnTableX(SqlOperationWithSeed):
```
Comment (philip-stoev) [inline misc/python/materialize/scalability/operations.py:70]: This is not currently used in any workload.

File: misc/python/materialize/scalability/operation.py [hunk 2]
```diff
@@ -35,20 +35,65 @@ def _execute(self, data: OperationData) -> OperationData:
...
 
         return data
 
+    def sql_statement_based_on_input(self, input: OperationData) -> str:
+        raise NotImplementedError
+
+
+class SqlOperationWithSeed(SqlOperationWithInput):
+    def __init__(self, seed_key: str):
+        self.seed_key = seed_key
```
Comment (philip-stoev) [inline misc/python/materialize/scalability/operation.py:61]: Is seed the right term here? Seems like it is actually a `parameter` . So `SqlOperationWithParameter` , etc?

**Review by philip-stoev — APPROVED** [review]

**Comment by nrainer-materialize** [comment]
Two builds so far with:
```
Benchmarking workload 'CreateAndDropTableWithMvWorkload' at concurrency 16 ...
==> mzcompose: test case workflow-default failed: <class 'psycopg.OperationalError'>: consuming input failed: server closed the connection unexpectedly
```

**Comment by nrainer-materialize** [comment]
Rebased to include https://github.com/MaterializeInc/materialize/commit/86ead0ad46b47560c290da435e0be98e7a7044a5.

**Review by ParkMyCar — APPROVED** [review]
Nice!

**Comment by philip-stoev** [comment]
14 hours is to much ... maybe this is a workload that should only be triggerable manually and be added to any automatic CIs only after progress on DDL scalability has been made?

**Comment by nrainer-materialize** [comment]
> 14 hours is to much ... maybe this is a workload that should only be triggerable manually and be added to any automatic CIs only after progress on DDL scalability has been made?

Absolutely, no question. I am playing around with the values (`count`, `exponent-base`, `max-concurrencies`) to find a suitable setup.

**Comment by nrainer-materialize** [comment]
> > 14 hours is to much ... maybe this is a workload that should only be triggerable manually and be added to any automatic CIs only after progress on DDL scalability has been made?
> 
> Absolutely, no question. I am playing around with the values (`count`, `exponent-base`, `max-concurrencies`) to find a suitable setup.

Down to 1:40h now with two workloads and this configuration:
```
              - --count
              - "128"
              - --exponent-base
              - "4"
              - --max-concurrency
              - "64"
```

Going to merge this one now.

**Pull request merged**